File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/ShadowWisp.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class ShadowWisp : Affliction
    {
        public override string Name { get { return "Shadow Wisp"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.Attacking, 
                };
            }
        }

        protected override void OnAttacking(ICharacter attacker, ICharacter target, ISpell spell, ref double damage, ref double modifier)
        {
            if (spell.SpellType == SpellType.Magical && RollPercentage(10))
            {
                var minion = SummonMinionCopyAmongEnemy(attacker, "Lingering Death");
                minion.Spells.Clear();
                AddSpell(minion, spell.Name);
                ApplyAffliction(minion, "Clearcasting");
                ApplyAffliction(minion, "Certain Death");
                SetVar(minion, "death_counter", 3);
            }
        }
    }
}

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/ShadowWisp.lua
function GetName()
  return "Shadow Wisp"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { Attacking }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == Attacking and param2:GetSpellType() == Magical and Roll(0, 99) < 10 then
    local wisp = SummonMinionCopy(char, "Shadow Wisp")
    wisp:ClearSpells()
    wisp:AddSpell(param2:GetName())
    wisp:ApplyAffliction("Clearcasting")
    wisp:ApplyAffliction("Certain Death")
    wisp:SetVar("death_counter", 3)
  end
end
